File: FountainPensNg.Server/Data/Models/Ink.cs
﻿namespace FountainPensNg.Server.Data.Models;
public class Ink : Entity {
    public int Id { get; set; }
    public string Maker { get; set; } = "";
    public string InkName { get; set; } = "";
    public string Comment { get; set; } = "";
    public string Photo { get; set; } = "";
    public string Color { get; set; } = "";
    public double? Color_CIELAB_L { get; set; }
    public double? Color_CIELAB_a { get; set; }
    public double? Color_CIELAB_b { get; set; }
    public int Rating { get; set; }
    public int Ml { get; set; }
    public string ImageObjectKey { get; set; } = "";
    public required NpgsqlTsVector FullText { get; set; }
    public virtual List<InkedUp>? InkedUps { get; set; }
}

File: fountainpensng.client/src/dtos/Ink.ts
import { FountainPen } from "./FountainPen";
import { InkedUpForListDTO } from "./InkedUpDTO";

export interface Ink {
  id: number;
  maker: string;
  inkName: string;
  comment: string;
  photo: string;
  color: string;
  ml: number;
  rating: number;
  inkedUps: InkedUpForListDTO[];
  currentPens: FountainPen[] | null;
  penDisplayName: string | null;
  imageObjectKey: string;
  imageUrl: string;
  cieLAB_Sort: number;
}
